internal/scheduler/schedulerobjects/resourcelist.go

```diff
@@ -33,13 +33,17 @@ func (a QuantityByPriorityAndResourceType) String() string {
 }
 
 func (a QuantityByPriorityAndResourceType) Add(b QuantityByPriorityAndResourceType) {
-	for p, rsb := range b {
-		rsa := a[p]
-		rsa.Add(rsb)
-		a[p] = rsa
+	for p, rlb := range b {
+		a.AddResouceList(p, rlb)
 	}
 }
 
+func (a QuantityByPriorityAndResourceType) AddResouceList(priority int32, rlb ResourceList) {
+	rla := a[priority]
```

Comment:
what happens if a[priority]  doesn't exist?

Reply:
It's initialised to the zero value, which is safe.